README.md

```diff
@@ -2,51 +2,170 @@
+### Insertion
+Content being served at a certain position.
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:user_info``` | UserInfo | Yes | The user info structure.
+```:insertion_id``` | String | Yes | Generated by the SDK (*do not set*)
+```:request_id``` | String | Yes | Generated by the SDK (*do not set*)
+```:content_id``` | String | No | Identifier for the content to be shown, must be set.
+```:properties``` | Properties | Yes | Any additional custom properties to associate. For v1 integrations, it is fine not to fill in all the properties.
+```:position``` | Number | Yes | Content location in the list of insertions, calculated by Promoted and uses the Paging parameters. (*do not set*)
+---
+### Paging
+#### TODO
+---
+### Request
+A request for content insertions.
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:user_info``` | UserInfo | Yes | The user info structure.
+```:request_id``` | String | Yes | Generated by the SDK (*do not set*)
+```:use_case``` | String | Yes | One of the use case values, i.e. 'FEED' (see [constants.rb](https://github.com/promotedai/promoted-ruby-client/blob/main/lib/promoted/ruby/client/constants.rb)).
+```:properties``` | Properties | Yes | Any additional custom properties to associate.
+```:paging``` | Paging | Yes | Paging parameters (see TODO)
+---
+### MetricsRequest
+Input to ```prepare_for_logging```
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:request``` | Request | No | The underlying request for content.
+```:full_insertion``` | [] of Insertion | No | The proposed list of insertions.
+---
 
-Or
+### DeliveryRequest
+Input to ```deliver```
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:experiment``` | CohortMembership | Yes | A cohort to evaluation in experimentation.
+```:request``` | Request | No | The underlying request for content.
+```:full_insertion``` | [] of Insertion | No | The proposed list of insertions with all metadata, will be compacted before forwarding to Promoted.
+```:only_log``` | Boolean | Yes | Defaults to false. Set to true to override whether Delivery API is called for this request.
+---
 
-1. Clone the repo on your local machine
-2. `cd promoted-ruby-client`
-3. `bundle`
-4. `irb -Ilib -rpromoted/ruby/client`
+### LogRequest
 
-A console will launch with the library loaded.  Here is example code to use.
+Output of ```prepare_for_logging``` as well as an ouput of an SDK call to ```deliver```, input to ```send_log_request``` to log to Promoted
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:request``` | Request | No | The underlying request for content to log.
+```:insertion``` | [] of Insertion | No | The insertions, which are either the original request insertions or the insertions resulting from a call to ```deliver``` if such call occurred.
```

Comment:
We chatted in person.  It'd be good to document the motivation and benefit of the clients.

Reply:
Agree. Let's chat offline about the best format/venue for that. I will add it to the issue I'm tracking.